staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -38,12 +43,15 @@ type AuditContext struct {
 	// RequestAuditConfig is the audit configuration that applies to the request
 	RequestAuditConfig RequestAuditConfig
```

Comment:
we're still reading / writing this unprotected ... it's hard to tell what goroutines the writes / reads are coming from ... is this field still a race panic risk?

Reply:
(and I guess if you pull in my POC from https://github.com/kubernetes/kubernetes/pull/129472#discussion_r2079865926, the same question about the Sink will apply)

Reply:
Ack. picked up your commits. will dig into the 2 TODO's you left for me!